app/src/forms/rbac/service.js

```diff
@@ -0,0 +1,440 @@
+      .modify('filterKeycloakId', params.keycloakId)
+      .modify('filterUsername', params.username)
+      .modify('filterFullName', params.fullName)
+      .modify('filterFirstName', params.firstName)
+      .modify('filterLastName', params.lastName)
+      .modify('filterEmail', params.email)
+      .modify('filterFormId', params.formId)
+      .modify('filterFormName', params.formName)
+      .modify('filterPublic', params.public)
+      .modify('filterActive', params.active)
+      .modify('filterArray', 'roles', params.permissions)
+      .modify('orderDefault');
+
+    const roles = await UserFormRoles.query()
+      .skipUndefined()
+      .modify('filterId', params.userId)
+      .modify('filterKeycloakId', params.keycloakId)
+      .modify('filterUsername', params.username)
+      .modify('filterFullName', params.fullName)
+      .modify('filterFirstName', params.firstName)
+      .modify('filterLastName', params.lastName)
+      .modify('filterEmail', params.email)
+      .modify('filterFormId', params.formId)
+      .modify('filterFormName', params.formName)
+      .modify('filterPublic', params.public)
+      .modify('filterActive', params.active)
+      .modify('filterArray', 'roles', params.permissions)
+      .modify('orderDefault');
+    return toFormUsers(permissions, roles);
+  },
+
+  getUserForms: async (params) => {
+    const permissions = await UserFormPermissions.query()
+      .skipUndefined()
+      .modify('filterId', params.userId)
+      .modify('filterKeycloakId', params.keycloakId)
+      .modify('filterUsername', params.username)
+      .modify('filterFullName', params.fullName)
+      .modify('filterFirstName', params.firstName)
+      .modify('filterLastName', params.lastName)
+      .modify('filterEmail', params.email)
+      .modify('filterFormId', params.formId)
+      .modify('filterFormName', params.formName)
+      .modify('filterPublic', params.public)
+      .modify('filterActive', params.active)
+      .modify('filterArray', 'roles', params.permissions)
+      .modify('orderDefault');
+
+    const roles = await UserFormRoles.query()
+      .skipUndefined()
+      .modify('filterId', params.userId)
+      .modify('filterKeycloakId', params.keycloakId)
+      .modify('filterUsername', params.username)
+      .modify('filterFullName', params.fullName)
+      .modify('filterFirstName', params.firstName)
+      .modify('filterLastName', params.lastName)
+      .modify('filterEmail', params.email)
+      .modify('filterFormId', params.formId)
+      .modify('filterFormName', params.formName)
+      .modify('filterPublic', params.public)
+      .modify('filterActive', params.active)
+      .modify('filterArray', 'roles', params.permissions)
+      .modify('orderDefault');
+
+    return toUserForms(permissions, roles);
+  },
+
+  setFormUsers: async (formId, userId, data) => {
+    // check this in middleware? 422 in valid params
+    if (!formId || 0 === formId.length) {
```

Comment:
Style: Normally in JS you check for equality by having the thing you are checking for be the first statement, and then the constant or condition to match as the second statement. Obviously not a hard/fast style guide and it's still obvious what's being done, but nearly all examples have it in reverse. (https://github.com/airbnb/javascript#control-statements)